### PR TITLE
Cleaned up cmd line options

### PR DESCRIPTION
**Pull request by improbable-danny**
Cleaned up the cmd line option handling code which will be useful for NFR pipeline without modifying the settings (+ added a few extra parameters) 

-------

#### Description
Modified how we handle command line args (internal only) - it makes it bit easier to add new args without copy/paste and exposed a few of the other settings 

#### Tests
Testing using both -Argument and -Argument=[True/False] method

STRONGLY SUGGESTED: How can this be verified by QA?
Test passing a few command line arguments and check the desired result 

#### Primary reviewers
@m-samiec

### CODE REVIEW

**Comment by MatthewSandfordImprobable** [comment]
Love it :) So clean *-*


Does make me think if we could do one parse of the command line and pull out the options into some CommandLineOptions class and then use that to set the gdk settings. It's only a small improvement though, just avoiding some extra string finds so meeh.

File: SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp [hunk 1]
```diff
@@ -123,34 +109,12 @@ void USpatialGDKSettings::PostInitProperties()
...
-	UE_LOG(LogSpatialGDKSettings, Log, TEXT("Use Is Actor Relevant For Connection is %s."), UseIsActorRelevantForConnection ? TEXT("enabled") : TEXT("disabled"));
-	UE_LOG(LogSpatialGDKSettings, Log, TEXT("Batch Spatial Position Updates is %s."), bBatchSpatialPositionUpdates ? TEXT("enabled") : TEXT("disabled"));
+	CheckCmdLineOverrideBool(CommandLine, TEXT("OverrideSpatialWorkerConnectionOnGameThread"), TEXT("Spatial worker connection on game thread"), bRunSpatialWorkerConnectionOnGameThread);
+	CheckCmdLineOverrideBool(CommandLine, TEXT("OverrideResultTypes"), TEXT("Result types"), bEnableResultTypes);
+	CheckCmdLineOverrideBool(CommandLine, TEXT("OverrideNetCullDistanceInterest"), TEXT("Net cull distance interest"), bEnableNetCullDistanceInterest);
+	CheckCmdLineOverrideBool(CommandLine, TEXT("OverrideNetCullDistanceInterestFrequency"), TEXT("Net cull distance interest frequency"), bEnableNetCullDistanceFrequency);
+	CheckCmdLineOverrideBool(CommandLine, TEXT("OverrideActorRelevantForConnection"), TEXT("Actor relevant for connection"), UseIsActorRelevantForConnection);
```
Comment (MatthewSandfordImprobable) [inline SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp]: Ahh not you issue really but maybe we can change this to bUseIsActorRelevantForConnection

File: SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp [hunk 2]
```diff
@@ -12,6 +12,26 @@
 
 DEFINE_LOG_CATEGORY(LogSpatialGDKSettings);
 
+namespace
+{
+	void CheckCmdLineOverrideBool(const TCHAR* CommandLine, const TCHAR* Parameter, const TCHAR* PrettyName, bool& Value)
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp]:
```suggestion
	void CheckCmdLineOverrideBool(const TCHAR* CommandLine, const TCHAR* Parameter, const TCHAR* PrettyName, bool& bOutValue)
```

File: SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp [hunk 3]
```diff
@@ -70,46 +90,12 @@ void USpatialGDKSettings::PostInitProperties()
...
-	}
-	UE_LOG(LogSpatialGDKSettings, Log, TEXT("Unreal load balancing is %s."), bEnableUnrealLoadBalancer ? TEXT("enabled") : TEXT("disabled"));
+	CheckCmdLineOverrideBool(CommandLine, TEXT("OverrideSpatialOffloading"), TEXT("Offloading"), bEnableOffloading);
+	CheckCmdLineOverrideBool(CommandLine, TEXT("OverrideServerInterest"), TEXT("Server interest"), bEnableServerQBI);
+	CheckCmdLineOverrideBool(CommandLine, TEXT("OverrideHandover"), TEXT("Handover"), bEnableHandover);
+	CheckCmdLineOverrideBool(CommandLine, TEXT("OverrideLoadBalancer"), TEXT("Load balancer"), bEnableUnrealLoadBalancer);
+	CheckCmdLineOverrideBool(CommandLine, TEXT("OverrideRPCRingBuffers"), TEXT("RPC ring buffers"), bUseRPCRingBuffers);
+	CheckCmdLineOverrideBool(CommandLine, TEXT("OverrideRPCPacking"), TEXT("RPC packing"), bPackRPCs);
 
 	if (bEnableUnrealLoadBalancer)
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp]: nit: move this to after the rest of the parsing

**Pull request merged**